.github/workflows/manual_deploy_infra.yml

```diff
@@ -63,7 +65,7 @@ jobs:
   approve:
     needs: plan
     runs-on: ubuntu-latest
-    environment: terraform
+    environment: terraform-${{ inputs.key }}
```

Comment:
I've got a minor concern that the environment name is used by GitHub for more than just labelling but happy to give this a go